core/src/main/java/org/geysermc/geyser/translator/protocol/java/inventory/JavaOpenBookTranslator.java

```diff
@@ -0,0 +1,78 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.geyser.translator.protocol.java.inventory;
+
+import com.github.steveice10.mc.protocol.data.game.inventory.ContainerType;
+import com.github.steveice10.mc.protocol.packet.ingame.clientbound.inventory.ClientboundOpenBookPacket;
+import com.github.steveice10.mc.protocol.packet.ingame.serverbound.inventory.ServerboundContainerClosePacket;
+import org.geysermc.geyser.inventory.GeyserItemStack;
+import org.geysermc.geyser.inventory.Inventory;
+import org.geysermc.geyser.item.Items;
+import org.geysermc.geyser.network.GameProtocol;
+import org.geysermc.geyser.session.GeyserSession;
+import org.geysermc.geyser.translator.inventory.InventoryTranslator;
+import org.geysermc.geyser.translator.protocol.PacketTranslator;
+import org.geysermc.geyser.translator.protocol.Translator;
+import org.geysermc.geyser.util.InventoryUtils;
+
+@Translator(packet = ClientboundOpenBookPacket.class)
+public class JavaOpenBookTranslator extends PacketTranslator<ClientboundOpenBookPacket> {
+
+    @Override
+    public void translate(GeyserSession session, ClientboundOpenBookPacket packet) {
+        GeyserItemStack stack = session.getPlayerInventory().getItemInHand();
+
+        // Don't spawn a fake lectern for books already opened "normally" by the client.
+        if (stack.getItemData(session).equals(session.getCurrentBook())) {
+            session.setCurrentBook(null);
+            return;
+        }
+
+        // Only post 1.20.60 is it possible to tell the client to open a lectern.
+        if (!GameProtocol.is1_20_60orHigher(session.getUpstream().getProtocolVersion())) {
+            return;
+        }
+
+        if (stack.asItem().equals(Items.WRITTEN_BOOK)) {
+            Inventory openInventory = session.getOpenInventory();
+            if (openInventory != null) {
+                InventoryUtils.closeInventory(session, openInventory.getJavaId(), true);
+
+                ServerboundContainerClosePacket closeWindowPacket = new ServerboundContainerClosePacket(openInventory.getJavaId());
+                session.sendDownstreamGamePacket(closeWindowPacket);
+            }
+
+            InventoryTranslator translator = InventoryTranslator.inventoryTranslator(ContainerType.LECTERN);
+            session.setInventoryTranslator(translator);
+
+            // Should never be null
+            assert translator != null;
+            Inventory inventory = translator.createInventory("", 69, ContainerType.LECTERN , session.getPlayerInventory());
```

Comment:
Try not to make this 69 (nice) a magic number - use constants or explain what's going on.